### PR TITLE
Feature/search page

### PR DESCRIPTION
**Pull request by K-w-e**
Here my first version of the search page, i kept it simple.
You can select/deselect to search transactions by type or in a specified account. (every filter is already enabled on initial state)

Let me know if you have any suggestions!

### CODE REVIEW

File: lib/providers/accounts_provider.dart [hunk 1]
```diff
@@ -11,11 +11,18 @@ final accountColorProvider = StateProvider<int>((ref) => 0);
...
 final countNetWorthSwitchProvider = StateProvider<bool>((ref) => true);
 final selectedAccountCurrentMonthDailyBalanceProvider = StateProvider<List<FlSpot>>((ref) => const []);
+final filterAccountProvider = StateProvider<Map<int, bool>>((ref) => {});
+
 
 class AsyncAccountsNotifier extends AsyncNotifier<List<BankAccount>> {
   @override
   Future<List<BankAccount>> build() async {
     ref.watch(mainAccountProvider.notifier).state = await _getMainAccount();
+    _getAccounts().then((value) => {
```
Comment (lucaantonelli) [inline lib/providers/accounts_provider.dart:21]: I think the _getAccounts() call is redundant, you can just put a return inside the first one after the for loop, it should work.

```
return _getAccounts().then((value) => {
      for(BankAccount account in value) {
        ref.watch(filterAccountProvider.notifier).state[account.id!] = true
      }
      return value;
});
```

**Review by lucaantonelli — CHANGES_REQUESTED** [review]

**Review by theperu — COMMENTED** [review]
Apart from what has been already pointed out it looks good to me. I would just leave it open a bit and see if Federico (UX/UI designer) has any suggestion, great work! 😁

**Comment by federicopozzato** [comment]
Great work! I just revisited a bit under the UI point of view. You can find it in the figma (one of the last frames, 6.6.0), but I leave you a screenshot here: 
image

**Comment by K-w-e** [comment]
PR updated
I'm adding one thing, the initial state of the filters is no longer the same. Now, initially, only Income/Outcome and the main bank account are selected. In my opinion, with the new interface, if everything was selected from the start, it wasn't clear.

Let me know!

**Comment by federicopozzato** [comment]
That default state should be that none of the filters is applied, so all should be empty, then you can filter by tapping in them and when they are active they become filled with blue.

Was this your doubt?

**Comment by K-w-e** [comment]
Oh ok, fixed it

**Review by theperu — APPROVED** [review]
LGTM, I think we can merge this 👍🏻

File: lib/pages/search_page/search_page.dart [hunk 2]
```diff
@@ -0,0 +1,257 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:sossoldi/providers/accounts_provider.dart';
+import 'package:sossoldi/providers/transactions_provider.dart';
+import '../../custom_widgets/transactions_list.dart';
+import '../../model/transaction.dart';
```
Comment (mikev-cw) [inline lib/pages/search_page/search_page.dart:6]: Can you fit the first 2 imports paths to our code guidelines?
Just do the same as the last 2 imports, with the absolute path

**Review by mikev-cw — CHANGES_REQUESTED** [review]
LGTM too, just fix this small thing, then you're done.

Also I'm not sure about letting all disabled by default is the clearest thing, but we can speak about that on next meetings, for now it's great!

File: lib/pages/search_page/search_page.dart (hunk 2 above)
Comment (K-w-e) [reply lib/pages/search_page/search_page.dart]: Sure

**Pull request merged**